.github/workflows/main.yml

```diff
@@ -7,25 +7,53 @@ on:
     branches: ["main"]
   pull_request:
     branches: ["main"]
+  workflow_dispatch:
+    inputs:
+      environment:
+        required: true
+        description: "Deployment environment"
+        type: environment
```

Comment:
The `type` for the `environment` input should be `string` instead of `environment`. GitHub Actions input types are typically `string`, `boolean`, or `choice`.